argus/recipes/cloud/windows.py

```diff
@@ -328,49 +335,42 @@ def prepare_cbinit_config(self, service_type):
         # Append service IP as a config option.
         address = self.pattern.format(util.get_local_ip())
         self._cbinit_conf.set_conf_value(name=self.config_entry,
-                                         value=address)
+                                         value=address,
+                                         section=self.config_group)
 
 
 class CloudbaseinitEC2Recipe(CloudbaseinitMockServiceRecipe):
     """Recipe for EC2 metadata service mocking."""
 
-    config_entry = "ec2_metadata_base_url"
+    config_group = util.EC2_SERVICE
     pattern = "http://{}:2000/"
 
 
 class CloudbaseinitCloudstackRecipe(CloudbaseinitMockServiceRecipe):
     """Recipe for Cloudstack metadata service mocking."""
 
-    config_entry = "cloudstack_metadata_ip"
-    pattern = "{}:2001"
-
-    def pre_sysprep(self):
-        super(CloudbaseinitCloudstackRecipe, self).pre_sysprep()
+    config_group = util.CLOUD_STACK_SERVICE
+    pattern = "http://{}:2001"
```

Comment:
Somehow yes, we could specify the value in the config file, but by doing so we will also have to specify the port in the `metadata_url` also, and later on slice the `metadata_url` to get the port on which the service mock should listen on. 
I think we could do it that way so that we eliminate the hard coded ports for the mock services.